Merge testing -> main

liaozhuzhu opened the pull request:
- add test suite (/pyTests)
- add documentation
- add partners

Commit:
Add Necessary Files for Submission

Commit:
Start flushing out readme

Commit:
Update testing file

Commit:
update readme

Commit:
poking around at nfa

Commit:
Merge branch 'main' of https://github.com/AmBha21/Theory-of-Computation into testing

Commit:
Update readme after merge

Commit:
Update readme and partner.txt

Commit:
Added test suite as well as further documentation for code/tests

Commit:
Added test suite as well as further documentation for code/tests

Commit:
add disclaimer to testing

AmBha21 commented:
Like I mentioned before, you need to test whether the setters of each datatype work. Also, import the regexp_to_nfa module so that you don't have to copy and paste the entire file over and over. Lastly, you could put all related tests (nfa tests, etc) in one file, similar to JUnit in software, so that a user can run all related tests at once or chose to run specific ones. I'm not sure if there's something like that in Python, but I feel like it has to be there.
One thing to note, the testing_stuff and everything in it was me experimenting with minimizing DFAs, so just add in. That way when we hackathon this Wednesday or something I'll have a good starting point. Also delete the generated pngs

Commit:
readd testing_stuff as testing_mini

Commit:
Made single file test suite for dfa / nfa

Commit:
Update test cases + add setter test cases

Commit:
update doc

liaozhuzhu commented:
> Like I mentioned before, you need to test whether the setters of each datatype work. Also, import the regexp_to_nfa module so that you don't have to copy and paste the entire file over and over. Lastly, you could put all related tests (nfa tests, etc) in one file, similar to JUnit in software, so that a user can run all related tests at once or chose to run specific ones. I'm not sure if there's something like that in Python, but I feel like it has to be there. One thing to note, the testing_stuff and everything in it was me experimenting with minimizing DFAs, so just add in. That way when we hackathon this Wednesday or something I'll have a good starting point. Also delete the generated pngs

- Added setter tests
- Encapsulated dfa / nfa tests into their own respective suites
- Added assert testing instead of normal comparison
- Added back minimization testing folder labeled as 'testing_mini'
